src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -257,58 +254,53 @@ internal static ProcessInfo[] GetProcessInfos(Predicate<int>? processIdFilter =
-                        buffer = new long[(bufferSize + 7) / 8];
-                    }
+                uint requiredSize = 0;
 
-                    uint requiredSize = 0;
-
-                    unsafe
+                unsafe
+                {
+                    // Note that the buffer will contain pointers to itself and it needs to be pinned while it is being processed
+                    // by GetProcessInfos below
+                    fixed (byte* bufferPtr = buffer)
                     {
-                        // Note that the buffer will contain pointers to itself and it needs to be pinned while it is being processed
-                        // by GetProcessInfos below
-                        fixed (long* bufferPtr = buffer)
+                        // some platforms require the buffer to be 64-bit aligned.
+                        // ArrayPool does not guarantee 64-byte alignment, so we take the address of first 64-byte aligned element
+                        // we round up the address of the pinned array to the nearest multiple of 64
+                        Debug.Assert(bufferSize > 64);
+                        byte* alignedBufferPtr = (byte*)(((nint)bufferPtr + 63) & ~63);
+                        int firstAlignedElementIndex = (int)(alignedBufferPtr - bufferPtr);
+
+                        uint status = Interop.NtDll.NtQuerySystemInformation(
+                            Interop.NtDll.SystemProcessInformation,
+                            alignedBufferPtr,
+                            (uint)(buffer.Length - firstAlignedElementIndex),
+                            &requiredSize);
+
+                        if (status != Interop.NtDll.STATUS_INFO_LENGTH_MISMATCH)
                         {
-                            uint status = Interop.NtDll.NtQuerySystemInformation(
-                                Interop.NtDll.SystemProcessInformation,
-                                bufferPtr,
-                                (uint)(buffer.Length * sizeof(long)),
-                                &requiredSize);
-
-                            if (status != Interop.NtDll.STATUS_INFO_LENGTH_MISMATCH)
+                            // see definition of NT_SUCCESS(Status) in SDK
+                            if ((int)status < 0)
                             {
-                                // see definition of NT_SUCCESS(Status) in SDK
-                                if ((int)status < 0)
-                                {
-                                    throw new InvalidOperationException(SR.CouldntGetProcessInfos, new Win32Exception((int)status));
-                                }
-
-                                // Parse the data block to get process information
-                                processInfos = GetProcessInfos(MemoryMarshal.AsBytes<long>(buffer), processIdFilter);
-                                break;
+                                ArrayPool<byte>.Shared.Return(buffer);
```

Comment:
Rather than guarding every potential throw in this function, why not just place everything in a try/finally?

Reply:
I've moved it to finally block